main.cpp

```diff
@@ -392,6 +477,35 @@ DoubleType& DoubleType::divide(double rhs)
     return *this;
 }
 
+DoubleType& DoubleType::pow(const DoubleType& exp)
+{
+    powInternal(double(exp));
+    return *this;
+}
+
+DoubleType& DoubleType::pow(const FloatType& exp)
+{
+    powInternal(static_cast<double>(float(exp)));
+    return *this;
+}
+
+DoubleType& DoubleType::pow(const IntType& exp)
+{
+    powInternal(static_cast<double>(int(exp)));
+    return *this;
+}
+
+DoubleType& DoubleType::pow(double exp)
+{
+    powInternal(exp);
+    return *this;
+}
```

Comment:
after you make `powInternal` chainable, revise these functions to return what `powInternal` returns.
- [ ] revise